geosnap/harmonize/harmonize.py

```diff
@@ -0,0 +1,161 @@
+import numpy as np
+import pandas as pd
+import geopandas as gpd
+from tobler.area_weighted import *
+from tobler.nlcd import _check_presence_of_crs
+
+def harmonize(raw_community, 
+              target_year_of_reference,
+              weights_method = 'area', 
+              extensive_variables = [], 
+              intensive_variables = [],
+              allocate_total = True,
+              raster = None,
+              codes = [21, 22, 23, 24],
+              force_crs_match = True):
+    """
+    Harmonize Multiples GeoData Sources with different approaches
+
+    Parameters
+    ----------
+
+    raw_community : tuple
+        Multiple GeoDataFrames given by geosnap.data.Community(..., type = "raw") (see (1) in Notes).
+    
+    target_year_of_reference : string
+        The target year that represents the bondaries of all datasets generated in the harmonization. Could be, for example '2010'.
+        
+    weights_method : string
+        The method that the harmonization will be conducted. This can be set to:
+            "area": harmonization according to area weights.
+            "nlcd_land_type_area" : harmonization according to the Land Types according to National Land Cover Data (NLCD)  considered 'populated' areas.
+            "nlcd_Poisson"        : NOT YET INTRODUCED.
+            "nlcd_Gaussian"       : NOT YET INTRODUCED.
+
+    extensive_variables : list
+        The names of variables in each dataset of raw_community that contains extensive variables to be harmonized (see (2) in Notes).
+        
+    intensive_variables : list
+        The names of variables in each dataset of raw_community that contains intensive variables to be harmonized (see (2) in Notes).
+    
+    allocate_total: boolean
+        True if total value of source area should be allocated.
+        False if denominator is area of i. Note that the two cases
+        would be identical when the area of the source polygon is
+        exhausted by intersections. See (3) in Notes for more details.
+        
+    raster : the associated NLCD raster (from rasterio.open) that has the types of each pixel in the spatial context.
+        Only taken into consideration for harmonization NLCD based.
+        
+    codes : an integer list of codes values that should be considered as 'populated' from the National Land Cover Database (NLCD).
```

Comment:
since we have other options for rasters that might be more applicable outside the US, could we make this terminology a little more generic? i.e. inputs are a raster, and a set of pixel values to consider, but not necessarily NLCD?

Reply:
Sure, I'll make the description more generic for the raster inputs (but, I think letting the default for the codes argument the same). Since I'll fix most of these things, I updated this PR to WIP, ok?

Reply:
👍 I think making the inputs more generic but keeping these defaults is perfect